pkg/common/telemetry/inmem_test.go

```diff
@@ -14,6 +21,92 @@ func TestNewInmemRunner(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestDefaultEnabledNewInmemRunner(t *testing.T) {
+	t.Run("enabled when block undeclared", func(t *testing.T) {
+		runner, err := newInmemRunner(testInmemConfig())
+		assert.Nil(t, err)
+		assert.True(t, runner.isConfigured())
+	})
+
+	t.Run("enabled flag undeclared", func(t *testing.T) {
+		config := testInmemConfig()
+		config.FileConfig = FileConfig{
+			InMem: &InMem{},
+		}
+		runner, err := newInmemRunner(config)
+		assert.Nil(t, err)
+		assert.True(t, runner.isConfigured())
+	})
+
+	t.Run("enabled flag declared", func(t *testing.T) {
+		enabledFlag := true
+
+		config := testInmemConfig()
+		config.FileConfig = FileConfig{
+			InMem: &InMem{
+				Enabled: &enabledFlag,
+			},
+		}
+		runner, err := newInmemRunner(config)
+		assert.Nil(t, err)
+		assert.True(t, runner.isConfigured())
+	})
+}
+
+func TestDisabledNewInmemRunner(t *testing.T) {
+	enabledFlag := false
+
+	config := &MetricsConfig{
+		ServiceName: "foo",
+		FileConfig: FileConfig{
+			InMem: &InMem{
+				Enabled: &enabledFlag,
+			},
+		},
+	}
+	runner, err := newInmemRunner(config)
+	assert.Nil(t, err)
+	assert.False(t, runner.isConfigured())
+}
+
+func TestWarnOnFutureDisable(t *testing.T) {
+	logger, hook := test.NewNullLogger()
+
+	// Get a real logrus.Entry
+	logger.SetLevel(logrus.DebugLevel)
+	logger.Debug("boo")
+	c := &MetricsConfig{
+		Logger:      hook.LastEntry(),
+		ServiceName: "foo",
+	}
+
+	ir, err := newInmemRunner(c)
+	require.Nil(t, err)
+	assert.Equal(t, 1, len(ir.sinks()))
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go ir.run(ctx)
+
+	// Wait for run to complete set up
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Sleeping in tests causes them to be either 1) artificially slow, or 2) flaky. It should be avoided.

What about something like this (pseudocode)?
```
	util.RunWithTimout(t, time.Minute, func() {
                log.Debug("sentinel") // see comment above
		for {
			// 1. send signal
                        // 2. check entry is not "sentinel" and has expected content
			// 3. check for context cancellation (indicating that we've timed out)
		}
	})
```

Reply:
see new pattern